tests/ios/xcodeproj/Single-Static-Framework-Project.xcodeproj/bazelinstallers/lldb-settings.sh

```diff
@@ -21,10 +21,39 @@ settings set target.sdk-path $SDKROOT
 settings set target.swift-framework-search-paths $FRAMEWORK_SEARCH_PATHS
 END
 
+# Holds all swift-extra-clang-flags
+LLDB_SWIFT_EXTRA_CLANG_FLAGS=()
+# Append extra clang flags if set
 if [[ -n ${BAZEL_LLDB_SWIFT_EXTRA_CLANG_FLAGS:-} ]]
+then
+  LLDB_SWIFT_EXTRA_CLANG_FLAGS+=($BAZEL_LLDB_SWIFT_EXTRA_CLANG_FLAGS)
+fi
+# Append the file 'BAZEL_LLDB_FRAMEWORK_SEARCH_PATHS_FILE' exists
+# it memans that there are framework search paths that LLDB should
+# be aware of so append the contents of that file
+#
+# Look for this logic in 'rules/xcodeproj.bzl'
+if [[ -n "$BAZEL_LLDB_FRAMEWORK_SEARCH_PATHS_FILE" ]]
+then
+  # All framework search paths from file
+  FSPS="$(cat $BAZEL_WORKSPACE_ROOT/$BAZEL_LLDB_FRAMEWORK_SEARCH_PATHS_FILE)"
+  # Xcode won't resolve these paths for us since we're
+  # passing it directly from file to file
+  #
+  # Manually solving 'BAZEL_WORKSPACE_ROOT' and 'PLATFORM_DIR' for now 
+  # until it's possible to pass bigger build settings to Xcode
+  #
+  # For context see: https://github.com/bazel-ios/rules_ios/pull/216
+  FSPS_WITH_ROOT_RESOLVED=${FSPS//"\$BAZEL_WORKSPACE_ROOT"/$BAZEL_WORKSPACE_ROOT}
+  FSPS_WITH_ROOT_RESOLVED=${FSPS_WITH_ROOT_RESOLVED//"\$(PLATFORM_DIR)"/$PLATFORM_DIR}
+  LLDB_SWIFT_EXTRA_CLANG_FLAGS+=($FSPS_WITH_ROOT_RESOLVED)
+fi
+# Set all swift-extra-clang-flags if 'LLDB_SWIFT_EXTRA_CLANG_FLAGS'
+# is not empty
+if [[ ${#LLDB_SWIFT_EXTRA_CLANG_FLAGS[@]} -ne 0 ]]
 then
   cat >> $BAZEL_LLDB_INIT_FILE <<-END
-settings set -- target.swift-extra-clang-flags $BAZEL_LLDB_SWIFT_EXTRA_CLANG_FLAGS
+settings set -- target.swift-extra-clang-flags ${LLDB_SWIFT_EXTRA_CLANG_FLAGS[@]}
```

Comment:
I was pondering Tulsi/XCHammer doesn't need this. I expect it is because we already wrote the search paths the xcode project.

Reply:
I'd definitely need some time kind of exploring the project generated by other tools to answer that question, not sure how lldb would work in this particular scenario

Reply:
I'm curious too. If `FRAMEWORK_SEARCH_PATHS` - if it's an argv problem then maybe you'll need the args file with said build.

That being said, perhaps pointing tools at a single directory would improve performance because the tools won't have so many directories to search - e.g. a single "Built products directory" or `/System/Library/Frameworks` exemplify this line of thinking.